skbio/metadata/tests/test_interval.py

```diff
@@ -0,0 +1,568 @@
+                                          'function': 'transport'})
+
+    def test_bad_constructor(self):
+        with self.assertRaises(TypeError):
+            Interval(interval_metadata=IntervalMetadata(),
+                     intervals=[1, (4, 7)],
+                     boundaries=[(True, False), (False, False)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+
+    def test_bad_constructor_bad_boundaries(self):
+        with self.assertRaises(ValueError):
+            Interval(interval_metadata=IntervalMetadata(),
+                     intervals=[(1, 2), (4, 7)],
+                     boundaries=[(False, False)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+
+    def test_repr_one_interval(self):
+        im = IntervalMetadata()
+        f = Interval(interval_metadata=im,
+                     intervals=[(1, 2)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+        exp = (r"<Interval: start=1, end=2, 1 contiguous interval,"
+               " 2 metadata keys, dropped=False>")
+        res = repr(f)
+        self.assertEqual(res, exp)
+
+    def test_repr_two_intervals(self):
+        im = IntervalMetadata()
+        f = Interval(interval_metadata=im,
+                     intervals=[(1, 2), (4, 7)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+        exp = (r"<Interval: start=1, end=7, 2 non-contiguous intervals,"
+               " 2 metadata keys, dropped=False>")
+        res = repr(f)
+        self.assertEqual(res, exp)
+
+    def test_str_two_intervals(self):
+        im = IntervalMetadata()
+        f = Interval(interval_metadata=im,
+                     intervals=[(1, 2), (4, 7)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+        exp = (r"<Interval: start=1, end=7, 2 non-contiguous intervals,"
+               " 2 metadata keys, dropped=False>")
+        res = str(f)
+        self.assertEqual(res, exp)
+
+    def test_cmp(self):
+        f1 = Interval(interval_metadata=IntervalMetadata(),
+                      intervals=[(1, 2)],
+                      boundaries=[(True, False)],
+                      metadata={'name': 'sagA', 'function': 'transport'})
+        f2 = Interval(interval_metadata=IntervalMetadata(),
+                      intervals=[(10, 20)],
+                      boundaries=[(True, False)],
+                      metadata={'name': 'sagA', 'function': 'transport'})
+        self.assertTrue(f1._cmp(f2))
+        self.assertFalse(f2._cmp(f1))
+
+    def test_contains(self):
+        f1 = Interval(interval_metadata=IntervalMetadata(),
+                      intervals=[(1, 2)],
+                      boundaries=[(True, False)],
+                      metadata={'name': 'sagA', 'function': 'transport'})
+        self.assertIn('name', f1.metadata)
+        self.assertIn('function', f1.metadata)
+        self.assertFalse('gene' in f1.metadata)
+
+    def test_hash(self):
```

Comment:
`hash` is also deceptive and can lead to very nasty bugs later. Would it be possible to add in tests for overlapping intervals, as well as enumerate and verify that changes in different components that make up the hash result in different hashes? It would be useful to explicitly test whether sort order of the intervals matters as well.